### ISSUE TITLE
install without NVCC

### ISSUE THREAD

**Issue by drcege** [AlexZhu2001/cuda_available#1]
When installed on a machine without a GPU, the CMake complains about the absence of nvcc.
```python
  × Building wheel for cuda-available (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [21 lines of output]
      WARNING: Use cmake.version instead of cmake.minimum-version with scikit-build-core >= 0.8
      *** scikit-build-core 0.8.1 using CMake 3.22.1 (wheel)
      *** Configuring CMake...
      2024-02-29 17:06:54,030 - scikit_build_core - WARNING - libdir/ldlibrary: /Users/null/miniconda3/lib/libpython3.9.a is not a real file!
      2024-02-29 17:06:54,030 - scikit_build_core - WARNING - Can't find a Python library, got libdir=/Users/null/miniconda3/lib, ldlibrary=libpython3.9.a, multiarch=darwin, masd=None
      loading initial cache file /var/folders/1s/wv8433110s98z3trwk4y9rrm0000gp/T/tmpsk513j9r/build/CMakeInit.txt
      CMake Error at /usr/local/Cellar/cmake/3.22.1/share/cmake/Modules/CMakeDetermineCUDACompiler.cmake:179 (message):
        Failed to find nvcc.
      
        Compiler requires the CUDA toolkit.  Please set the CUDAToolkit_ROOT
        variable.
      Call Stack (most recent call first):
        CMakeLists.txt:2 (project)
      
      
      CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
      -- Configuring incomplete, errors occurred!
      See also "/var/folders/1s/wv8433110s98z3trwk4y9rrm0000gp/T/tmpsk513j9r/build/CMakeFiles/CMakeOutput.log".
      See also "/var/folders/1s/wv8433110s98z3trwk4y9rrm0000gp/T/tmpsk513j9r/build/CMakeFiles/CMakeError.log".
      
      *** CMake configuration failed
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for cuda-available
Failed to build cuda-available
```

It is exactly because it is unknown whether the target system is equipped with a GPU that such tools are needed to check. Therefore, the tool should be installable on purely CPU systems, rather than assuming CUDA is definitely available.

**Comment by AlexZhu2001** [comment]
Sorry, I replied late. It seems that you are using the source code for installation, but the inability to find NVCC caused a compilation error.
In fact, the NVCC(cuda toolkit) does not require you to actually install a graphics card on the machine. For example, my GitHub action release environment does not have a graphics card and can still generate whl packages normally.
Also It is impossible to compile code without using NVCC or skip the compilation of NVCC, as NVIDIA does not provide a method to check the CUDA version and abilities using regular cpp code.
You can just install cuda toolkit without GPU installed in Linux.

**Comment by AlexZhu2001** [comment]
But it reminds me there is another issue that this code could failed if there is no cuda dylib installed, I will test it later.

**Comment by AlexZhu2001** [comment]
This issue was fixed in 0.3.0 by using pure python code dynamic load nvcuda.dll or libcuda.so.1 to query cuda device info
